modules/optional-enabling-read-only-mode-backup-restore.adoc

```diff
@@ -0,0 +1,357 @@
+# ...
+----
+
+. Distribute the new configuration bundle to all {productname} instances. 
+
+. Start {productname} by entering the following command:
++
+[source,terminal]
+----
+$ podman run -d --rm -p 80:8080 -p 443:8443  \
+   --name=quay-main-app \
+   -v $QUAY/config:/conf/stack:Z \
+   -v $QUAY/storage:/datastorage:Z \
+   {productrepo}/{quayimage}:{productminv}
+----
+
+. After starting {productname}, a banner inside in your instance informs users that {productname} is running in read-only mode. Pushes should be rejected and a 405 error should be logged. You can test this by running the following command:
++
+[source,terminal]
+----
+$ podman push <quay-server.example.com>/quayadmin/busybox:test
+----
++
+Example output
++
+[source,terminal]
+----
+613be09ab3c0: Preparing 
+denied: System is currently read-only. Pulls will succeed but all write operations are currently suspended.
+----
+
+[id="quay-ocp-readonly"]
+== Configuring read-only mode {productname-ocp}
+
+After the service keys have been created and added to your PostgreSQL database, you must restart the `Quay` container on your {ocp} deployment.
+
+[IMPORTANT]
+====
+Deploying {productname-ocp} in read-only mode requires you to modify the secrets stored inside of your {ocp} cluster. It is highly recommended that you create a backup of the secret prior to making changes to it. 
+====
+
+.Prerequisites
+
+* You have created the service keys and added them to your PostgreSQL database.
+
+.Procedure
+
+. Enter the following command to read the secret name of your {productname-ocp} deployment:
++
+[source,terminal]
+----
+$ oc get deployment -o yaml <quay_main_app_deployment_name>
+----
+
+. Use the `base64` command to encode the `quay-readonly.kid` and `quay-readonly.pem` files:
++
+[source,terminal]
+----
+$ base64 -w0 quay-readonly.kid
+----
++
+Example output
++
+[source,terminal]
+----
+ZjUyNDFm...
+----
++
+[source,terminal]
+----
+$ base64 -w0 quay-readonly.pem 
+----
++
+Example output
++
+[source,terminal]
+----
+LS0tLS1CRUdJTiBSU0E...
+----
+
+. Obtain the current configuration bundle and secret by entering the following command:
++
+[source,terminal]
+----
+$ oc get secret quay-config-secret-name -o json | jq '.data."config.yaml"' | cut -d '"' -f2 | base64 -d -w0 > config.yaml
+----
+
+. Edit the `config.yaml` file and add the following information:
++
+[source,yaml]
+----
+# ...
+REGISTRY_STATE: readonly
+INSTANCE_SERVICE_KEY_KID_LOCATION: 'conf/stack/quay-readonly.kid'
+INSTANCE_SERVICE_KEY_LOCATION: 'conf/stack/quay-readonly.pem'
+# ...
+----
+
+. Save the file and `base64` encode it by running the following command:
++
+[source,terminal]
+----
+$ base64 -w0 quay-config.yaml
+----
+
+. Scale down the {productname} Operator pods to `0`. This ensures that the Operator does not reconcile the secret after editing it.
++
+[source,terminal]
+----
+$ oc scale --replicas=0 deployment quay-operator.{productminv} -n openshift-operators
```

Comment:
**Here after updated the config secret managed by Quay Operator**, need to scale out Quay Operator, when Quay Customers want to put Quay to Normal State, remove following config from the config secret managed by Quay Operator, and let Quay Operator to reconcile the change.
```
REGISTRY_STATE: readonly
INSTANCE_SERVICE_KEY_KID_LOCATION: 'conf/stack/quay-readonly.kid'
INSTANCE_SERVICE_KEY_LOCATION: 'conf/stack/quay-readonly.pem'
```